docs/guides/backends.rst

```diff
@@ -0,0 +1,184 @@
+==============
+
+Backends are the interfaces used to run :ref:`Circuits <circuits_guide>` in Orquestra Core. In this guide we'll cover differences between types of backends (``QuantumBackend`` vs ``QuantumSimulator``, ``TrackingBackend``, ``SymbolicSimulator``), how to use backends, currently integrated backends, and creating your own backend (among other topics).
+
+
+General Backend Information
+===========================
+
+The function of a backend is to send instructions to quantum hardware or to a simulator. Since there are various types of quantum hardware and simulators, different sets of instructions needs to be written and it can become complex. ``QuantumBackend`` simplifies this process by enabling the user to deploy a single set of instructions on multiple hardware and simulators simultaneously. Therefore executing code on various devices becomes easier and more manageable.
+
+Two types of backends
+---------------------
+
+``QuantumBackend`` is the base class for all backends. ``QuantumBackend``\ s are used for physical quantum hardware backends.
+
+Because ``QuantumBackend``\ s connect to a cloud provider to run remote circuits, it is necessary to specify which device you want to connect to. For example, the ``QiskitBackend`` takes in a ``device_name`` string:
+
+.. literalinclude:: /examples/backends_guide.py
+  :start-after: # QuantumBackend creation example
+  :end-before: # End QuantumBackend creation example
+
+.. _quantum_simulator:
+
+``QuantumSimulator`` is a subclass of ``QuantumBackend`` with added functionality that only simulators can provide, such as getting the wavefunction of a circuit and getting exact expectation values. Unlike ``QuantumBackend``\ s, ``QuantumSimulator``\ s usually do not require a ``device_name`` parameter. One exception to this rule is the ``QiskitSimulator``, which has :ref:`multiple options <integrated_backends>`:
+
+.. literalinclude:: /examples/backends_guide.py
+  :start-after: # QuantumSimulator creation example
+  :end-before: # End QuantumSimulator creation example
+
+For more information on how to use each type of backend please refer to the :ref:`section in this guide on various methods <backend_methods>` or to the :doc:`API documentation </api/quantum/api/backend/index>`.
+
+Testing the Backends
+--------------------
+
+The `backend tests <https://github.com/zapatacomputing/orquestra-quantum/blob/main/src/orquestra/quantum/api/backend_test.py>`_ cover both general behaviors we want to make sure the ``Backend``\ s follow and specific gate tests. For instance, a common "general behavior" test across multiple backends is the ``run_circuitset_and_measure`` test. Here are examples in the `Cirq tests <https://github.com/zapatacomputing/orquestra-cirq/blob/main/tests/orquestra/integrations/cirq/simulator/simulator_test.py#L52=>`_ and the `Qiskit tests <https://github.com/zapatacomputing/orquestra-qiskit/blob/main/tests/orquestra/integrations/qiskit/backend/backend_test.py#L162=>`_.
```

Comment:
I'd add an intro paragraph here, sth along these lines:

"One of our goals with the QuantumBackends was to make sure that all the backends behave in a uniform way. In order to ensure that, we have introduced a suite of tests which all backends need to pass."

Reply:
fixed in 4551c621e5c9c2b2295716c545eb6211b836df64 and also moved testing the backends to be a subsection of integrating the backends (I think that makes sense!)